File: concatenator-ocaml/lib/concatenator.ml
(* concat routine *)
let concat_files path extension exclude_folders =
  let default_filters =
    File_manager.get_concatenator_filters extension exclude_folders
  in
  let files = File_manager.walk_files path ~filter:default_filters in
  let read_files = File_manager.read_files files in
  String.concat "\n" read_files

File: concatenator-hs/src/Concatenator.hs
module Concatenator (concatFiles) where
    
import CommandLine (CommandLineOptions(..))
import FileManager (walkFiles, defaultConcatenatorFilters, readFileToString)
import Logger (getNewLogger)
import System.Log.Logger (logL)
import System.Log (Priority(INFO))
import Control.Monad (forM)

concatFiles :: CommandLineOptions -> IO String
concatFiles opts = do
    fileContents <- getFiles opts
    return $ concat fileContents

getFiles :: CommandLineOptions -> IO [String]
getFiles opts = do
    let directoryPath = directory opts
        extn = extension opts
        excludeFldrs = excludeFolders opts
        filters = defaultConcatenatorFilters extn excludeFldrs

    logger <- getNewLogger "concatenator"
    logL logger INFO ("Concatenating files: " ++ directoryPath)

    walked <- walkFiles filters directoryPath
    contents <- forM walked readFileToString
    return $ concat contents



